### PR TITLE
[WIP] initiall version of PR test jobs

### LINKED ISSUE #10: Run functional testsuite for PR builds
Related to https://github.com/Tendrl/tendrl-ci/issues/6.

We would like to be able to install cluster and run functional test suite for every PR before it will be merged.

Intended workflow would be:
Once PR will be ready for merge, somebody will manually trigger test job for the respective PR
* The job will be triggered via comment `run tests` in the respective PR
* ❔ It might be possible to somehow specify dependencies on PRs from other repos (probably in the PR description or the `run tests...` comment).

### PR DESCRIPTION
**Pull request by dahorak**
Fixes https://github.com/Tendrl/tendrl-ci/issues/10

Jobs for launching test suite for every PR.

It will create one job for each repository.
The job might be triggered by `run tests` phrase in the respective PR.

It is possible to specify dependencies on PRs from other repos this way:

``Depends on: tendrl-ui/pr809 tendrl-notifier/pr153``

either in the PR description or in the comment triggering the test execution (or on both places).

It has to be on one line without any additional formatting and has to follow quite precisely this format, to be able to correctly (and easily) parse it.
Basically the line has to start with `depends on:` string (case insensitive) followed by space separated list of dependencies in the form of: `<reponame>/pr<id>`, where `<reponame>` is the name of the repository (including `tendrl-` prefix) and `<id>` is the PR number.

### CODE REVIEW

**Comment by dahorak** [comment]
The Jobs are visible on [tendrl-PR CentOS CI Jenkins](https://ci.centos.org/view/tendrl-PR/) page and are named `Tendrl test (PR) - <repo-name>`.

**Comment by r0h4n** [comment]
@TimothyAsir Please review this

File: jobs/build-triggers/tendrl-build-triggers-pr-tests.yml [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+          recipients: dahorak@redhat.com
+
+
+- project:
+    name: 'Test_suite'
+    triggers:
+      - github-pull-request:
+          admin-list:
+            - dahorak
+          white-list:
```
Comment (cybernth) [inline jobs/build-triggers/tendrl-build-triggers-pr-tests.yml:115]: Is there a way to input this rather than hardcoding

File: jobs/build-triggers/tendrl-build-triggers-pr-tests.yml [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+          echo
+          echo "##############################################################"
+          echo "Run whole test suite."
+          echo
+      - trigger-builds:
+          - project: "tendrl-2-cluster-gluster"
+            property-file: env.sh
+            block: true
+
+    publishers:
```
Comment (cybernth) [inline jobs/build-triggers/tendrl-build-triggers-pr-tests.yml:104]: Where the test results will be posted?

**Review by cybernth — CHANGES_REQUESTED** [review]

File: jobs/build-triggers/tendrl-build-triggers-pr-tests.yml (hunk 1 above)
Comment (dahorak) [reply jobs/build-triggers/tendrl-build-triggers-pr-tests.yml:115]: It's the same as for the build jobs. Itshould be possible, to automatically fetch the list of people from the organization specified below in the org-list section, but there seems to be some problem with this and I'm not sure why it doesn't work and where might be the problem with that.
I'll try to check that once more and find some more information about that - that will be definitely the best solution.

File: jobs/build-triggers/tendrl-build-triggers-pr-tests.yml (hunk 2 above)
Comment (dahorak) [reply jobs/build-triggers/tendrl-build-triggers-pr-tests.yml:104]: The overall status of the installation and tests is reported back to the PR where it was triggered (as a `CentOS CI - tests` line in the *checks* section) and *Details* link points to the particular job.
More granular results are available in the subsequent jobs (see *Subproject Builds* section of the Jenkins job).

Additionally when it fails, it will notify me over email, so I can check if it is real installation or test failure, or if it is problem with the tests itself.
